src/hotspot/os/linux/cgroupSubsystem_linux.cpp

```diff
@@ -627,54 +627,66 @@ void CgroupSubsystemFactory::cleanup(CgroupInfo* cg_infos) {
  * number of active processors in the system.
  *
  * If quotas have been specified, the resulting number
- * returned will never exceed the number of active processors.
+ * set in the result reference will never exceed the number
+ * of active processors.
  *
  * return:
- *    number of CPUs
+ *    true if there were no errors. false otherwise.
  */
-int CgroupSubsystem::active_processor_count() {
-  int quota_count = 0;
+bool CgroupSubsystem::active_processor_count(int& value) {
   int cpu_count;
-  int result;
+  int result = -1;
```

Comment:
Why not get rid of result and use `value` throughout like you did in the cached case?